packages/blockchain/src/blockchain.ts

```diff
@@ -367,15 +367,10 @@ export class Blockchain implements BlockchainInterface {
           throw new Error(
             'Cannot put a different genesis block than current blockchain genesis: create a new Blockchain',
           )
+          // genesis block is not optimistic
+          optimistic = false
         }
 
-        const { header } = block
-        const blockHash = header.hash()
-        const blockNumber = header.number
-        let td = header.difficulty
```

Comment:
this diff has primarily shifted into the else part of the optimistic condition on L414 with some basic initializations still out side the if {} else {} block